packages/ioc/src/container.ts

```diff
@@ -82,8 +82,18 @@ export class Container {
 
         const existing = this.__singletons.get(key);
         if (!existing) {
+            // @jacoobes should we return false if no existing key?
             throw Error("No existing key to swap for " + key);
         }
+        // check if there's dispose hook, and call it
+        if (hasCallableMethod(existing, 'dispose')) {
+            this.disposeAll().then(() => {;
```

Comment:
this.disposeAlll would call dispose on all objects, not intended

Reply:
This was what I was stuck on, how would we call dispose on existing object, delete it from hooks array, and then add existing?

Reply:
will see